src/encoder.rs

```diff
@@ -1232,8 +1232,9 @@ pub fn encode_tx_block<T: Pixel>(
   );
 
   let eob = ts.qc.quantize(coeffs, qcoeffs, tx_size, tx_type);
+  let has_coeff: bool = eob != 0;
 
-  let has_coeff = if need_recon_pixel || rdo_type.needs_coeff_rate() {
+  if rdo_type.needs_coeff_rate() {
```

Comment:
I think you have missed the condition "need_recon_pixel ||" here? That become true, for ex, when there are multiple tx blocks in a partition and the prediction mode of the partition is intra prediction. Since ref pixels for there should be pixel domain.